lib/avromatic/model/allowed_writer_methods_memoization.rb

```diff
@@ -0,0 +1,16 @@
+module Avromatic
+  module Model
+    module AllowedWriterMethodsMemoization
+      def self.included(base)
+        base.class_attribute :virtus_object_allowed_writer_methods
+        base.prepend(ClassMethods)
+      end
+
+      module ClassMethods
```

Comment:
I think we should rename this `InstanceMethods`. The module that is being prepended is overriding instance methods. After that I think this is good to go.